90-rtc-sys-time-init.rules: Optimize

valentindavid opened the pull request:
Two optimization here:
* We filter `IMPORT{cmdline}`. This causes some system calls. On core20, it also reads UEFI variables. This is very slow. We used to read the cmdline on all device. Now we do it only on rtc devices.
* We stop calling `/usr/bin/systemd-escape`.

The first optimization is the most effective part. On google cloud, we reduce the time to cold boot devices from around 30s to 4s.

valentindavid commented:
![Bootchart before fix](https://github.com/snapcore/core20/assets/432830/0158b862-c4e1-4ee0-bfde-6f62e7585248)

![Bootchart after fix](https://github.com/snapcore/core20/assets/432830/8df8b44b-8dfd-42b9-acca-5089ffbfe412)

ernestl commented on static/usr/lib/core/rtc-sys-time-init:
```diff
@@ -40,6 +40,11 @@ journal() {
 
+# ... or a path in /sys
+if ! [ -c "${dev_node}" ]; then
+    dev_node="$(udevadm info --query=property --path "${dev_node}" | sed "/^DEVNAME=/{;s///;q;};d")"
```
The ';' after the q seems redundant

valentindavid replied:
Fixed.

ernestl reviewed (approved):
Nice optimisation!

valentindavid commented on static/usr/lib/udev/rules.d/90-rtc-sys-time-init.rules:
```diff
@@ -1,12 +1,18 @@
 # instantiate rtc-sys-time-init@.service for the device node
-IMPORT{cmdline}="ubuntu_core.rtc_sys_time_init"
-ACTION=="add", SUBSYSTEM=="rtc", KERNEL=="rtc*", ATTR{hctosys}=="1", ENV{ubuntu_core.rtc_sys_time_init}!="?*", TAG+="systemd", PROGRAM="/usr/bin/systemd-escape --path %N", ENV{SYSTEMD_WANTS}+="rtc-sys-time-init@%c.service"
+IMPORT{cmdline}=="ubuntu_core.rtc_sys_time_init"
```
To avoid warning messages on old versions of systemd

Meulengracht reviewed (approved):
Looks super nice! A description for the PR would be nice though.

alfonsosanchezbeato commented:
Out of curiosity, why was it that udevd was reading UEFI variables? And why it changed in more modern versions?

valentindavid commented:
> Out of curiosity, why was it that udevd was reading UEFI variables? And why it changed in more modern versions?

It was added to udev by mistake.

Introduced in https://github.com/systemd/systemd/commit/53aa0d02add93d8c0afa2772609372a2040c162a

And removed in https://github.com/systemd/systemd/commit/09835de3ccfee80703d00cd16978de2c296706ed

It is supposed to be used for systemd options. Not for anything else.

alfonsosanchezbeato commented:
> > Out of curiosity, why was it that udevd was reading UEFI variables? And why it changed in more modern versions?
> 
> It was added to udev by mistake.
> 
> Introduced in [systemd/systemd@53aa0d0](https://github.com/systemd/systemd/commit/53aa0d02add93d8c0afa2772609372a2040c162a)
> 
> And removed in [systemd/systemd@09835de](https://github.com/systemd/systemd/commit/09835de3ccfee80703d00cd16978de2c296706ed)
> 
> It is supposed to be used for systemd options. Not for anything else.

Ha, interesting, thanks for finding out.
